docs/learn/encyclopedia/contract-development/types/fully-typed-contracts.mdx

```diff
@@ -69,29 +66,54 @@ Options:
   -h, --help  Print help
 ```
 
-Like any other CLI, you can also get help for any of these subcommands using something like `native balance --help`. Stellar CLI again fetches the on-chain interface types, this time using it to generate a full list of all arguments to the function, and even generates examples.
+Like any other CLI, you can also get help for any of these subcommands using something like `… -- balance --help`. Stellar CLI again fetches the on-chain interface types, this time using it to generate a full list of all arguments to the function, and even generates examples.
 
 :::tip
 
-And just to be totally clear, you don't need to alias these commands. This would work, too:
+If you're unfamiliar with the `--` double dash separator, this is a pattern used by other CLIs. Everything after the double dash, sometimes called the [slop](https://github.com/clap-rs/clap/issues/971), gets passed to the child process. An example of other CLIs that make use of this are `npm run` and `cargo run`.
 
-```bash
-soroban contract invoke --network testnet --id CDLZFC3SYJYDZT7K67VZ75HPJVIEUVNIXF47ZG2FB2RMQQVU2HHGCYSC -- balance --help
+:::
+
+<!-- The following paragraph is also outdated, but I'm not sure the most up-to-date PRs/Issues/Discussions to link to. Maybe we can just remove this paragraph? -->
+Of course, the exact way that that Stellar CLI parses arguments is an ongoing design discussion. Representing complex custom arguments on the command line is a design challenge no blockchain CLI gets perfect yet; many, including Stellar CLI, currently use JSON. But this has downsides, and other options are possible. [How would you like to see it work?](https://docs.google.com/document/d/1LASaLyxIA2-YqLnn2NqI5NZGnEmdzx9lLh-j7n0-i4A/edit#)
```

Comment:
Re-adding my comment here, to make it easier to read:

> The following paragraph is also outdated, but I'm not sure the most up-to-date PRs/Issues/Discussions to link to. Maybe we can just remove this paragraph?

Reply:
I would think the stellar/stellar-cli repo would be the relevant resource to link to, if we wanted. But, I think removing the paragraph is probably the best route.

Reply:
Done